src/Client/Grpc/GrpcDurableTaskClient.cs

```diff
@@ -78,10 +78,11 @@ public override async Task<string> ScheduleNewOrchestrationInstanceAsync(
     {
         Check.NotEntity(this.options.EnableEntitySupport, options?.InstanceId);
 
+        // We're explicitly OK with an empty version from the options as that had to be explicitly set. It should take precedence over the default.
         string version = string.Empty;
-        if (!string.IsNullOrEmpty(orchestratorName.Version))
+        if (options != null && options.Version != null)

```

Comment:
I think pattern matching may be beneficial here? Not sure if I have the syntax right.
```suggestion
        if (options is { Version: v })
```

or null-operator:

``` CSharp
if (options?.Version is { } v)
```